Modules/IO/TIFF/src/itkTIFFReaderInternal.h

```diff
@@ -66,6 +66,9 @@ class ITKIOTIFF_HIDDEN TIFFReaderInternal
   float    m_XResolution;
   float    m_YResolution;
   uint16_t m_SampleFormat;
+
+  bool m_WarningSilence{ false };
```

Comment:
It might be useful to add public API to suppress warnings/errors in this reader.

Reply:
That would be another PR. This PR is just to add the feature to use reentrant messages to the ITK loggers when the TIFF feature is available.